src/proxy/reconnect.rs

```diff
@@ -169,9 +174,21 @@ where
     type Future = <Reconnect<M, T> as svc::Service<Req>>::Future;
 
     fn poll_ready(&mut self) -> Poll<(), Self::Error> {
+        fn apply_jitter(duration: Duration, max_jitter: f64) -> Duration {
+            let jitter = rand::random::<f64>() * max_jitter;
+            let secs = (duration.as_secs() as f64) * jitter;
+            let nanos = (duration.subsec_nanos() as f64) * jitter;
+            let millis = (secs * 1000f64) + (nanos / 1000000f64);
```

Comment:
Consider using [mul_add](https://doc.rust-lang.org/std/primitive.f64.html#method.mul_add) here especially since this is something that could be executed quite frequently!